app/controllers/application_controller.rb

```diff
@@ -860,19 +860,12 @@ def get_reports_menu(group = current_group, tree_type = "reports", mode = "menu"
     folders = []
     user = current_user
     @sb[:grp_title] = reports_group_title
-    @data = []
+    data = []
     if !group.settings || group.settings[:report_menus].blank? || mode == "default"
       # array of all reports if menu not configured
-      @rep = MiqReport.for_user(current_user).sort_by { |r| [r.rpt_type, r.filename.to_s, r.name] }
-      if tree_type == "timeline"
-        @data = @rep.reject { |r| r.timeline.nil? }
-      else
-        @data = @rep.select do |r|
-          r.template_type == "report" && !r.template_type.blank?
-        end
-      end
-      @data.each do |r|
-        next if r.template_type != "report" && !r.template_type.blank?
+      data = MiqReport.for_user(current_user).where(:template_type => "report").order(:rpt_type, :filename, :name)
```

Comment:
Ah, it *is* possible to have a `nil` filename.  I'm not sure what that will do to ordering, but it would likely have some impact.  None of the tests are based on order :smile: so, I'm getting away with it there.

Not sure if the ending data will be impacted.

I'll do some queries locally and see what that looks like.